fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2381,6 +2381,9 @@ def _post_state_dict_hook(
     # recursively, so we need to make sure that we only clone each tensor at
     # most once. Thus we add an attribute on the tensor called "_has_been_cloned"
     # which keeps track of tensors that are no longer at risk of being freed.
+    if dist.get_rank() != 0:
```

Comment:
Will this be a backwards incompatible change?

Reply:
This is a bug fix to avoid OMM, it shouldn't impact correctness or user facing behavior.

Reply:
Sure, I understand the OOM issue. Let me rephrase my question :) Previously, we were returning a full state dict on all ranks. With this chance we will be returning an empty state dict on some of the ranks when users call `state_dict`.  Is that correct? Would something break if users execute something based on this assumption?

Reply:
Thank you for the explanation. Please correct me if I'm wrong. If user call `_post_state_dict_hook` on a rank, this will be a breaking change. My understanding is that `_post_state_dict_hook` is for colon tensor and copy from rank 0 is sufficient?  Do we have internal meta use case or other oss use case that user calls `_post_state_dict_hook` on a rank?

Reply:
Can you explain this "colon tensor and copy from rank 0 is sufficient?" statement?  _post_state_dict_hook is called every time a user calls state_dict on the model. This is a post processing function in a sense. 
Can we add a flag to make this optional? We can then think of this being the default in the future?

Reply:
>Can you explain this "colon tensor and copy from rank 0 is sufficient?" statement?

It was from reading the comments of this function.  I was curious that if function's goal is to copies tensor, does only copy from rank 0 be enough?

>Can we add a flag to make this optional? We can then think of this being the default in the future?
Yeah make sense, will do!